presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1165,6 +1165,14 @@ public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context)
 
         QualifiedName name = getQualifiedName(context.qualifiedName());
 
+        boolean ignoreNulls = false;
+        if (context.nullTreatment() != null) {
+            String functionName = name.toString();
+            check(functionName.equalsIgnoreCase("lag") || functionName.equalsIgnoreCase("lead"),
```

Comment:
This check needs to happen in the analyzer. It should throw `SemanticException(NOT_SUPPORTED)`